migration/scripts/migration.js

```diff
@@ -0,0 +1,108 @@
+const admin = require('firebase-admin');
+const serviceAccountUSA = require('./env.production-usa.json');
+const serviceAccountCA = require('./env.production-canada.json');
+
+admin.initializeApp({
+  credential: admin.credential.cert(serviceAccountCA),
+});
+const firestore = admin.firestore();
+
+// Reference new Firestore database
+const customersDB = firestore.collection('customers');
+const placesDB = firestore.collection('places');
+const sensorsDB = firestore.collection('sensors');
+const systemsDB = firestore.collection('systems');
+
+// Reference local data (exported from legacy realtime databse)
+const customersData = require('../data/demoCustomer.json');
+const legacyData = require('../data/legacyData-shared.json');
+
+// Categorise data into collections
+const placesData = legacyData.places;
+const sensorsData = legacyData.sensors;
+const systemsData = legacyData.systems;
+
+// Set customers data
+const addCustomers = (id, newDoc) => customersDB
+  .doc(id)
+  .set(newDoc)
+  .catch((error) => {
+    console.error(error);
+  });
+
+// Object.keys(customersData).forEach((key) => {
+//   addCustomers(key, customersData[key]);
+// });
+
+// Set places data
+const addPlaces = (id, newDoc) => placesDB
+  .doc(id)
+  .set(newDoc)
+  .catch((error) => {
+    console.error(error);
+  });
+
+Object.keys(placesData).forEach((key) => {
+  // Delete dead data
+  delete placesData[key].admins;
+  delete placesData[key].sensors;
+  placesData[key].customerId = '';
+  addPlaces(key, placesData[key]);
+});
+
+// Set sensors data
+const addSensors = (id, newDoc) => sensorsDB
+  .doc(id)
+  .set(newDoc)
+  .catch((error) => {
+    console.error(error);
+  });
+
+Object.keys(sensorsData).forEach((key) => {
+  sensorsData[key].customerId = '';
+  // Make systems object an array of systemIds
+  addSensors(key, sensorsData[key]);
+  if (sensorsData[key].systems) {
+    Object.keys(sensorsData[key].systems).forEach((systemId) => {
+      sensorsData[key].systemIds = [systemId];
+    });
+  }
+  // Add datachain as a sub-collection of each sensor doc in sensorsData
+  if (sensorsData[key].datachain) {
+    const datachainRef = firestore.collection('sensors').doc(`${key}`);
+    const { datachain } = sensorsData[key];
+    Object.keys(datachain).forEach((datachainKey) => {
+      const addSubcollection = () => datachainRef
+        .collection('datachain')
```

Comment:
What's the benefit of having these as a subcollection vs array? (Genuinely asking, not asking to changing the code)

Reply:
I will double check with Kei on this but we decided to put this in a subcollection to give the database a little more structure (some sensors have a lot of information in the datachain) and to set the datachain up to be more scalable for future.